### PR TITLE
Fix mp_radix_size off by 1 error

### PR DESCRIPTION
**Pull request by embhorn**
The 'size' set by `mp_radix_size` should conditionally add one when WC_DISABLE_RADIX_ZERO_PAD is defined and the zero padding would be added. This issue could lead to a buffer overflow when passed to `mp_toradix` (or other API allocating a buffer based on the size being string length + 1).

This issue was reported in ZD10901.

### CODE REVIEW

**Comment by SparkiDev** [comment]
Why is it off by one?
Is it a leading zero being added to make up the byte? (If so there is a define to turn this off I believe.)

**Comment by embhorn** [comment]
It's a counting error, I believe. Instead of counting the number of characters starting with one, it starts with zero. At the end it adds 1 to supposedly handle the NULL terminator, but then all (well most, `wolfSSL_BN_bn2dec` did not adjust and was susceptible to this problem) of the functions that call it still have to adjust by one byte to allow for the NULL at the end. Even if the number is a single digit (or 0) it still has a size (strlen) of 1 + 1 (for NULL).

**Review by SparkiDev — CHANGES_REQUESTED** [review]
The problem is the leading zero.
Detect if the leading zero is going to be added and add one.
WC_DISABLE_RADIX_ZERO_PAD disables this, so don't check if this is defined.

**Comment by embhorn** [comment]
That makes sense. Updated `mp_radix_size` to conditionally add one when WC_DISABLE_RADIX_ZERO_PAD is defined and the zero padding would be added.

**Comment by SparkiDev** [comment]
When number is negative, this code won't work.

File: wolfcrypt/src/tfm.c [hunk 1]
```diff
@@ -5483,6 +5483,13 @@ int mp_radix_size (mp_int *a, int radix, int *size)
     }
     fp_zero (t);
 
+#ifndef WC_DISABLE_RADIX_ZERO_PAD
+    /* For hexadecimal output, add zero padding when number of digits is odd */
+    if ((digs & 1) && (radix == 16)) {
```
Comment (SparkiDev) [inline wolfcrypt/src/tfm.c:5488]: This won't work when the number is negative.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/tfm.c (hunk 1 above)
Comment (embhorn) [reply wolfcrypt/src/tfm.c]: Good catch, thanks Sean.

File: wolfcrypt/src/tfm.c [hunk 2]
```diff
@@ -5483,6 +5483,13 @@ int mp_radix_size (mp_int *a, int radix, int *size)
     }
     fp_zero (t);
 
+#ifndef WC_DISABLE_RADIX_ZERO_PAD
+    /* For hexadecimal output, add zero padding when number of digits is odd */
+    if (((a->sign == FP_NEG) ? !(digs & 1) : (digs & 1)) && (radix == 16)) {
```
Comment (SparkiDev) [inline wolfcrypt/src/tfm.c:5488]: You can check for negative after this.
And fix integer.c as well.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

**Review by SparkiDev — APPROVED** [review]

**Review by toddouska — APPROVED** [review]

**Pull request merged**